### DISCUSSION TITLE
Animations

### DISCUSSION THREAD

**Discussion by jweintraub** [twilio-labs/paste#1660] [General] [answered]
Is there any general guidance for animations, optimal speeds, etc? We're planning to show an animation in the nav when we go in and out of a service nav.

**Comment by gloriliale** [comment]
Hi @jweintraub! So, we haven't tackled animation guidelines as part of Paste yet, but could you give us some visuals around what you're planning, and what you mean by going in and out of a service nav? We could offer specific support!

**Comment by jweintraub** [comment, marked as answer]
Hi! Please see attached screencast to see what I mean for going in and out of a service nav. We'd like the service nav to slide in from right when entering a service and slide out from right when going back to main nav. Any guidelines you can provide are helpful but if you don't have any yet, I can make up something that looks good.

https://user-images.githubusercontent.com/13891347/125495295-5f3f8a70-266b-4ad1-9fde-f5c573dc07be.mov

](url)

**Reply by gloriliale** [reply]
Thanks for sending this over @jweintraub!
The first thing that jumps out is that this service nav completely covers the previous menu- is there a way to preserve the hierarchy of Messaging --> Services so there isn't a loss of context if a user looks away?

For the animation itself, we have a slide animation in `@twilio-paste/animation-library` which has spring physics that would work. It's a result of a user click, so our suggestion would be that it be quick and last from 100-200 ms. 

You could also consider a mutation effect, where opening up Services bounces open the Services nav, moves it up under Messaging, and has everything else fade away, as opposed to the whole previous menu disappearing. Example of the bounce open and close part of that is[ the menus on this page](https://react-spring.io/).

**Reply by jweintraub** [reply]
Thank you so much. This is very helpful!

**Comment by jweintraub** [comment]
Can you share the link to @twilio-paste/animation-library? Not able to find it.

**Reply by andioneto** [reply]
Hi @jweintraub 👋 `@twilio-paste/animation-library` is a wrapper around React Spring, the third party animation library we use for Paste. 
- [`@twilio-paste/animation-library` source code](https://github.com/twilio-labs/paste/tree/main/packages/paste-libraries/animation)
- [React Spring documentation](https://react-spring.io/)
